### PR TITLE
mysql sink: add a batch size limiter

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
add a batch size limiter for mysql.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -347,6 +361,26 @@ func (s *mysqlSink) concurrentExec(ctx context.Context, rowGroups map[string][]*
 	return eg.Wait()
 }
 
+func txnRowLimiter(rows []*model.RowChangedEvent, maxTxnRow int) [][]*model.RowChangedEvent {
+	evaluateResultLength := (len(rows) / maxTxnRow) + 1
+	result := make([][]*model.RowChangedEvent, 0, evaluateResultLength)
```
Comment (amyangfei) [inline cdc/sink/mysql.go]: Can we only maintain a slice, which holds the index of RowChangedEvent we will consume each time, such as 10 events, batch size is 4, we get a slice of [4, 8, 10] and we consume rows[:4], rows[4:8], rows[8:10]. Then we save one time to copy some pointers.

File: cdc/sink/mysql.go [hunk 2]
```diff
@@ -347,6 +361,26 @@ func (s *mysqlSink) concurrentExec(ctx context.Context, rowGroups map[string][]*
...
+	for len(rows) > maxTxnRow {
+		lastTs := rows[maxTxnRow-1].Ts
+		i := maxTxnRow
+		for ; i < len(rows); i++ {
+			if lastTs < rows[i].Ts {
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/sink/mysql.go]: yes, The purpose of this design is to maintain transaction consistency

**Pull request merged**